File: src/main/webapp/js/choice.js
/*global Ext */
Ext.operon.tutorials.Choice=Ext.extend(Ext.util.Observable,
{
	constructor:function(question,data)
	{	
		Ext.operon.tutorials.Choice.superclass.constructor.call(this);
		this.choice_id=data.id;
		//this.choice_id=data.choice_id;
		this.letter=null;
		this.text=data.text;
		this.response=data.response;
		this.correct=data.correct;
		
		// state
		this.showresponse=false;	
		this.selected=false;
		this.current=false;
		
		// stats
		/*
		stats:
		{
			count: 0,
			fraction: '10/500'
		}
		*/
	}
});

File: src/main/java/org/operon/tutorials/Choice.java
package org.operon.tutorials;

public class Choice extends Item
{
	private String text="text";
	private String response="response";
	private Boolean correct=false;

	public Choice(){}
	
	public Choice(String id)
	{
		this.id=id;
	}

	public String getText(){return this.text;}
	public void setText(String text){this.text=text;}

	public String getResponse(){return this.response;}
	public void setResponse(String response){this.response=response;}

	public Boolean getCorrect(){return this.correct;}
	public void setCorrect(Boolean correct){this.correct=correct;}
}
